tracer_options.go

```diff
@@ -95,6 +95,14 @@ func (tracerOptions) Extractor(format interface{}, extractor Extractor) TracerOp
 }
 
 func (tracerOptions) Observer(observer Observer) TracerOption {
+	old := oldObserver{obs: observer}
+
+	return func(tracer *Tracer) {
+		tracer.observer.append(&old)
+	}
```

Comment:
Ok, but delegation will require scope resolution here, since, `ContribObserver` is also an interface and a `tracerOptions` function. Are you ok with having this (asking this because it doesn't conform to the current convention for other functions in tracer_options.go)?
```
func (t *tracerOptions) Observer(observer Observer) TracerOption {
    return t.ContribObserver(&oldObserver{obs: observer})
}
```